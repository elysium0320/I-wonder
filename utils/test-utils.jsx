import { configureStore } from '@reduxjs/toolkit';
import { render as rtlRender } from '@testing-library/react';
import React from 'react';
import { Provider } from 'react-redux';

import app from '../src/redux/reducers/app';

function render(
  ui,
  {
    preloadedState,
    store = configureStore( { reducer: { app }, preloadedState } ),
    ...renderOptions
  } = {}
) {
  function Wrapper( { children } ) {
    return <Provider store={ store }>{ children }</Provider>
  }

  return rtlRender( ui, { wrapper: Wrapper, ...renderOptions } )
};

export * from '@testing-library/react';

export { render };
