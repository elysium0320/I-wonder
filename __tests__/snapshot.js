import { render } from '../utils/test-utils';

import Index from '../pages/index';

it( 'renders index unchanged', () => {
  const { container } = render( <Index /> );

  expect( container ).toMatchSnapshot();
} );
