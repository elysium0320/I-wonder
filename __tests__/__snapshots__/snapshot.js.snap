// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders index unchanged 1`] = `
<div>
  <div
    class="Index"
  >
    <div
      class="indexContainer"
    >
      <div
        class="Logo stacked"
        data-testid="Logo"
      >
        <div
          class="imageContainer"
        >
          <img
            src="/images/spongebob.png"
          />
        </div>
        <h1>
          I Wonder
        </h1>
      </div>
      <div
        class="Tagline Tagline"
        data-testid="Tagline"
      >
        <p>
          Do you even remember what the issue is
        </p>
        <p>
          You just trying to find where the tissue is
        </p>
        <p
          class="fst-italic"
        >
          I Wonder, Kanye West
        </p>
      </div>
      <form
        class="Search Search"
        data-testid="Search"
      >
        <div
          class="form-floating"
        >
          <input
            class="form-control "
            data-testid="input"
            id="githubUrl"
            placeholder="Github URL"
            type="text"
          />
          <label
            data-testid="label"
            for="githubUrl"
          >
            Github URL
          </label>
          <img
            src="/icons/search.svg"
          />
        </div>
        <div
          class="buttonContainer"
        >
          <button
            class="btn btn-primary"
            data-testid="submit"
            type="submit"
          >
            View issues
          </button>
          <button
            class="btn btn-outline-primary"
            data-testid="lucky"
            type="button"
          >
            Feeling lucky
          </button>
        </div>
      </form>
    </div>
  </div>
</div>
`;
